numpy/core/tests/test_conversion_utils.py

```diff
@@ -107,18 +109,23 @@ class TestSearchsideConverter(StringConverterTestCase):
     """ Tests of PyArray_SearchsideConverter """
     conv = mt.run_searchside_converter
     def test_valid(self):
-        self._check('left', 'NPY_SEARCHLEFT')
-        self._check('right', 'NPY_SEARCHRIGHT')
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning,
+                       "partial matches and case insensitive matches")
+            self._check('left', 'NPY_SEARCHLEFT')
+            self._check('right', 'NPY_SEARCHRIGHT')
 
 
 class TestOrderConverter(StringConverterTestCase):
     """ Tests of PyArray_OrderConverter """
     conv = mt.run_order_converter
     def test_valid(self):
-        self._check('c', 'NPY_CORDER')
-        self._check('f', 'NPY_FORTRANORDER')
-        self._check('a', 'NPY_ANYORDER')
-        self._check('k', 'NPY_KEEPORDER')
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "case insensitive matches")
```

Comment:
This should be inside `self._check`

Reply:
Thanks @eric-wieser  for the review! i was considering that, but i didnt want this to suppress unintended warnings when TestSortKindConverter or TestSelectkindConverter is used. Since we are already filtering on DeprecatedWarning with specific message its probably better to put it back in self._check.

Reply:
If you put it in `self.check`, you can make it conditional on the value of `self.exact_match`.